pkg/apis/awsproviderconfig/v1alpha1/register.go

```diff
@@ -74,7 +74,7 @@ func NewCodec() (*AWSProviderConfigCodec, error) {
 }
 
 // DecodeProviderConfig deserialises an object from the provider config
-func (codec *AWSProviderConfigCodec) DecodeProviderConfig(providerConfig *clusterv1.ProviderConfig, out runtime.Object) error {
+func (codec *AWSProviderConfigCodec) DecodeProviderConfig(providerConfig *clusterv1.ProviderSpec, out runtime.Object) error {
```

Comment:
The variable name (`providerConfig`) should change too.

Reply:
As long as we don't remove providerConfig completely, providerConfig = providerSpec. I plan to change the naming globally as pointed out in the PR description.

Reply:
`providerConfig` changed to `providerSpec`